### ISSUE TITLE
Close inappbrowser on deeplink

### ISSUE THREAD

**Issue by BenjaminSAILcloud** [proyecto26/react-native-inappbrowser#312]
### Which platform(s) does your issue occur on?
- Both 
- All versions
- emulator
 
### Please, provide the following version numbers that your issue occurs with:

- CLI: 
```
react-native-cli: 2.0.1
react-native: 0.64.0
```
- Plugin(s): 
```
"react-native-inappbrowser-reborn": "3.6.3"
```

Hi,

I am using the inappbrowser to access a social website and I am NOT using it for authentication. In this website, there are buttons that links back to the app using deeplinks. I know they are working because when I access the website in browsers (e.g. Safari or Chrome) on the device, the deeplinks opens the app on the correct page. However when using the inappbrowser nothing happens.

How can I make the inappbrowser to handle correctly deeplinks without using the openAuth() method ?

Thanks

**Comment by BenjaminSAILcloud** [comment]
Hi,

Uppon more investigation, I found out that the deeplinks work correctly (e.g. they open the correct screen using react navigation) however when clicking the deeplink, the inappbrowser modal does not close...

**Comment by andriiDatsiuk** [comment]
@BenjaminSAILcloud take a look at a console of the page that should redirect user back to the app, probably you will see something like this: "Navigation is blocked: ...". I'm facing this issue as well. Here's what I've found so far.

https://bugs.chromium.org/p/chromium/issues/detail?id=738724
https://www.py4u.net/discuss/629990

The issue breaks auth flows with some 3rd party in my case. Users are not redirected back to the app after authentication.

**Comment by jdnichollsc** [comment]
Hello folks, please share a reproducible demo project to be able to debug your issue 👍

**Comment by sgtpsibin** [comment]
@jdnichollsc  Have you ever tried to make complete example of using openAuth(). I read alot but everyone say that we can not redirect to a deep link. (to close inappbrowser)
Do you have any example of redirect url can redirect to a deeplink or trigger universal link as well?

**Comment by mongkuen** [comment]
> Hello folks, please share a reproducible demo project to be able to debug your issue 👍

Appreciate the attention

Issue reproduced here: https://github.com/mongkuen/RNInAppBrowser-deeplink-demo

Video of issue
https://www.loom.com/share/26355f2b3c874c50960055275637041a

**Comment by TheWirv** [comment]
I second this. We have payment integration via InAppBrowser and have been using `openAuth` up until now, but that always triggers the OS dialog, "{App name} Wants to Use {payment provider URL} to Sign In", which is really annoying, and confusing to end users. I have tried just replacing the `openAuth` with `open`, but after that the redirect after successful payment just simply doesn't work anymore. E.g. PayPal is just "processing" forever, whereas before it redirected back to the app afterwards.
I have tried [this comment](https://github.com/proyecto26/react-native-inappbrowser/issues/306#issuecomment-1004399977), but that also does not work for me at all somehow. I tried putting a `console.log` inside the event listener, but nothing appears.

@jdnichollsc Is there a reason why the redirection logic has only been implemented for `openAuth`?

Oh, and on Android everything works fine. The Custom Tabs knows how to redirect out of the box.

**Comment by mongkuen** [comment]
This is probably a `SFSafariViewController` concern only. 

Not an iOS/Objective C dev, and this solution is in swift, but I imagine [this approach here](https://stackoverflow.com/a/38127120) might be close to the desired behavior? 

Then maybe a documentation section on how to modify `AppDelegate.m`/`AppDelegate.h` to support view dismissal?

**Comment by TheWirv** [comment]
Yeah, something's up here. The `SFSafariViewController` just won't redirect. I have tried opening the payment process in Safari view the bottom right button that `SFSafariViewController` provides. After payment was successful, Safari without any problem redirects to the app, or at least it asks whether to "open the URL in the app". But inside of `SFSafariViewController` there's just nothing happening.

**Comment by jdnichollsc** [comment]
> I second this. We have payment integration via InAppBrowser and have been using `openAuth` up until now, but that always triggers the OS dialog, "{App name} Wants to Use {payment provider URL} to Sign In", which is really annoying, and confusing to end users. I have tried just replacing the `openAuth` with `open`, but after that the redirect after successful payment just simply doesn't work anymore. E.g. PayPal is just "processing" forever, whereas before it redirected back to the app afterwards. I have tried [this comment](https://github.com/proyecto26/react-native-inappbrowser/issues/306#issuecomment-1004399977), but that also does not work for me at all somehow. I tried putting a `console.log` inside the event listener, but nothing appears.
> 
> @jdnichollsc Is there a reason why the redirection logic has only been implemented for `openAuth`?
> 
> Oh, and on Android everything works fine. The Custom Tabs knows how to redirect out of the box.

We're using the same strategy of WebBrowser (Expo), please check [here](https://docs.expo.dev/versions/latest/sdk/webbrowser/#webbrowseropenbrowserasyncurl-browserparams) for more details

**Comment by alexisloiselle** [comment]
Hey @TheWirv, I wanted to let you know that I got [the comment you referenced](https://github.com/proyecto26/react-native-inappbrowser/issues/306#issuecomment-1004399977) to work by adding this to the `AppDelegate.swift`:
```swift
  func application(_ app: UIApplication, open url: URL, options: [UIApplication.OpenURLOptionsKey : Any] = [:]) -> Bool {
    return RCTLinkingManager.application(app, open: url, options: options)
  }
```
Hope it helps!

**Comment by focux** [comment]
FYI, if you just want to close the in-app browser after handling a deep link, you just need to do:
```typescript
InAppBrowser.close()
```

In your `Linking.addEventListener` callback.
